src/main/java/com/ocelotslovebirds/birdhaus/datagen/RecipesGen.java

```diff
@@ -31,26 +30,26 @@ public RecipesGen(DataGenerator generatorIn) {
      * @param planks         Ingredient for the birdhouse.
      */
     private static void birdhouseRecipe(
-            Consumer<FinishedRecipe> recipeConsumer, ItemLike birdhouse, Tag log, Tag planks) {
+            Consumer<FinishedRecipe> recipeConsumer, ItemLike birdhouse, ItemLike log, ItemLike planks) {
         //Note: To add more ingredients, overload this method with more ItemLikes, provide the item in the function call
         //and provide a pattern and define below. Unlocked by controls what unlocks the recipe for a player in survival
         // mode.
 
         // Added tag item type - to enable spawing birdhouse using any type of plank OR logs
-        ShapedRecipeBuilder.shaped(birdhouse)
-                .pattern("LLL")
-                .pattern(" P ")
-                .pattern(" P ")
-                .define('L', log)
-                .define('P', planks)
-                .unlockedBy("has_log", has(log))
-                .unlockedBy("has_planks", has(planks))
-                .group("birdhouse")
-                .save(recipeConsumer);
+        ShapedRecipeBuilder shaped = ShapedRecipeBuilder.shaped(birdhouse);
+        shaped.pattern("LLL");
+        shaped.pattern(" P ");
+        shaped.pattern(" P ");
+        shaped.define('L', log);
+        shaped.define('P', planks);
+        shaped.unlockedBy("has_log", has(log));
+        shaped.unlockedBy("has_planks", has(planks));
+        shaped.group("birdhouse");
+        shaped.save(recipeConsumer);
     }
 
     @Override
     protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
-        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), BlockTags.LOGS, BlockTags.PLANKS);
+        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), Items.OAK_LOG, Items.OAK_PLANKS);
```

Comment:
why are we doing this?

Reply:
This led me on an adventure! I didn't know why initially, only that it was the recommended fix. So the reason is that minecraft made 'has' a private function that no longer can unlock a recipe based on tags, so all the tag-based unlocks for data-gen broke even though the recipes themselves still work fine. I went and readded that function and will upload a version that still takes tags shortly.